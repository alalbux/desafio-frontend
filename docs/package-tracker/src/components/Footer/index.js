import Grid from '@mui/material/Grid';
import Typography from '@mui/material/Typography';
import Link from '@/components/Link';
import Container from '@mui/material/Container';
import Copyright from '@/components/Copyright';
import LINKS from '@/locales/footer';

const Footer = () => {
  return (
    <Container component="footer" sx={styles.footerContainer}>
      <Grid container justifyContent="space-evenly">
        {LINKS.map((item) => (
          <Grid item key={item.title} sx={styles.gridItem}>
            <Typography {...styles.linkTypography}>{item.title}</Typography>
            <ul>
              {item.description.map((i) => (
                <li key={i}>
                  <Link text={i} sx={styles.linkList}></Link>
                </li>
              ))}
            </ul>
          </Grid>
        ))}
      </Grid>
      <Copyright sx={{ mt: 5 }} />
    </Container>
  );
};

const styles = {
  footerContainer: {
    mt: 8,
    py: [3, 6],
  },
  gridItem: {
    xs: 5,
    sm: 2,
  },
  linkTypography: {
    variant: 'h6',
    color: 'text.primary',
    gutterBottom: true,
    sx: {
      mx: 0,
      my: 2,
    },
  },
  linkList: {
    mx: 0,
    my: 2,
  },
};

export default Footer;
