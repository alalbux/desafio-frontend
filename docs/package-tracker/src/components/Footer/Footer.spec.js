import React from 'react';
import { render } from '@testing-library/react';
import Footer from '@/components/Footer';

describe('Footer', () => {
  it('should match the snapshot', () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
