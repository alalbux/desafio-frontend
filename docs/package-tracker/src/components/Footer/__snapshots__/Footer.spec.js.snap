// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match the snapshot 1`] = `
<DocumentFragment>
  <footer
    class="MuiContainer-root MuiContainer-maxWidthLg css-1ks4v4o-MuiContainer-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-hwacae-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-elk0q-MuiGrid-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-93ilsc-MuiTypography-root"
        >
          Para você
        </h6>
        <ul>
          <li>
            <a
              class="MuiTypography-root MuiTypography-button MuiLink-root MuiLink-underlineNone css-7ei8qr-MuiTypography-root-MuiLink-root"
              data-testid="link"
            >
              Entregador via aplicativo
            </a>
          </li>
        </ul>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-elk0q-MuiGrid-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-93ilsc-MuiTypography-root"
        >
          Para empresas
        </h6>
        <ul>
          <li>
            <a
              class="MuiTypography-root MuiTypography-button MuiLink-root MuiLink-underlineNone css-7ei8qr-MuiTypography-root-MuiLink-root"
              data-testid="link"
            >
              Entregador para escritórios
            </a>
          </li>
          <li>
            <a
              class="MuiTypography-root MuiTypography-button MuiLink-root MuiLink-underlineNone css-7ei8qr-MuiTypography-root-MuiLink-root"
              data-testid="link"
            >
              Soluções para varejo
            </a>
          </li>
          <li>
            <a
              class="MuiTypography-root MuiTypography-button MuiLink-root MuiLink-underlineNone css-7ei8qr-MuiTypography-root-MuiLink-root"
              data-testid="link"
            >
              Entrega para e-commerces
            </a>
          </li>
        </ul>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-elk0q-MuiGrid-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-93ilsc-MuiTypography-root"
        >
          Para entregar
        </h6>
        <ul>
          <li>
            <a
              class="MuiTypography-root MuiTypography-button MuiLink-root MuiLink-underlineNone css-7ei8qr-MuiTypography-root-MuiLink-root"
              data-testid="link"
            >
              Seja uma Transportadora Leve
            </a>
          </li>
          <li>
            <a
              class="MuiTypography-root MuiTypography-button MuiLink-root MuiLink-underlineNone css-7ei8qr-MuiTypography-root-MuiLink-root"
              data-testid="link"
            >
              Seja um entregador
            </a>
          </li>
        </ul>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-elk0q-MuiGrid-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-93ilsc-MuiTypography-root"
        >
          Loggi
        </h6>
        <ul>
          <li>
            <a
              class="MuiTypography-root MuiTypography-button MuiLink-root MuiLink-underlineNone css-7ei8qr-MuiTypography-root-MuiLink-root"
              data-testid="link"
            >
              Trabalhe conosco
            </a>
          </li>
        </ul>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-elk0q-MuiGrid-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-93ilsc-MuiTypography-root"
        >
          Dúvidas
        </h6>
        <ul>
          <li>
            <a
              class="MuiTypography-root MuiTypography-button MuiLink-root MuiLink-underlineNone css-7ei8qr-MuiTypography-root-MuiLink-root"
              data-testid="link"
            >
              loggi.com/ajuda
            </a>
          </li>
          <li>
            <a
              class="MuiTypography-root MuiTypography-button MuiLink-root MuiLink-underlineNone css-7ei8qr-MuiTypography-root-MuiLink-root"
              data-testid="link"
            >
              Imprensa
            </a>
          </li>
        </ul>
      </div>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-lgj0d-MuiTypography-root"
    >
      Copyright © 
      <a
        class="MuiTypography-root MuiTypography-button MuiLink-root MuiLink-underlineNone css-17f0go1-MuiTypography-root-MuiLink-root"
        data-testid="link"
      />
       2023.
    </p>
  </footer>
</DocumentFragment>
`;
